File: boats_to_save_people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        left = 0
        right = len(people) - 1
        boat = 0
        people.sort()

        while left <= right:
            boat += 1

            if people[left] + people[right] <= limit:
                left += 1
            right -= 1
        
        return boat

File: boats_to_save_people.java
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int left = 0;
        int right = people.length - 1;
        int boat = 0;
        Arrays.sort(people);

        while (left <= right) {
            boat += 1;

            if (people[left] + people[right] <= limit) {
                left += 1;
            }
            right -= 1;
        }

        return boat;        
    }
}
